### DISCUSSION TITLE
`自定义顶栏` > `布局设置>`的`元素呈现`是必须在合适的网络条件下才能完成首次加载吗？

### DISCUSSION THREAD

**Discussion by Geeyun-JY3** [the1812/Bilibili-Evolved#4727] [提问 / Q&A] [answered]
注意到在某标签页首次打开`自定义顶栏`设置项的`布局设置>`时，其中的`元素呈现`似乎是要在合适的网络条件下才能加载出来，如正式版 (GitHub 源) 和预览版须翻墙。

这样做是为什么呢？开这个讨论主要是感觉现在这样用户体验稍微差了那么一丢丢，网络条件没问题时也要等待约 1 秒才能加载出来。

如果是底层相关无能为力可直接关闭本讨论。

**Comment by the1812** [comment, marked as answer]
因为依赖一个第三方的排序库. 考虑到修改这个排序并不是高频操作, 所以设计成打开时再加载, 以前是放在本体里每次打开网页都加载比较浪费性能.

**Reply by Geeyun-JY3** [reply]
好的吧。

**Reply by tansongqi22** [reply]
事实上，不登录这个功能也无法加载
